core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -1693,4 +1700,42 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite with PrivateMethodT
       ShuffleBlockId(0, 5, 2), ShuffleBlockId(0, 6, 2)))
   }
 
+  test("logging of slow block fetches") {
+    val thresholdBps = 1024
+    val thresholdMs = 1000
+    val fetcher = createShuffleBlockIteratorWithDefaults(
+      Map.empty,
+      slowFetchThresholdMs = thresholdMs,
+      slowFetchThresholdBps = thresholdBps)
+    val totalBytes = 10000
+    val blockCount = 3
+    val bmId = BlockManagerId("execID", "hostname", 1234)
+
+    // Match the logger name used by the Logging trait
+    val logCapture = new Log4jCapture(ShuffleBlockFetcherIterator.getClass.getName.stripSuffix("$"))
+    try {
+      // Fetch with a read that is slightly slower than the threshold
+      val elapsedMs = 1000 * totalBytes / thresholdBps + 1
+      fetcher.logFetchIfSlow(elapsedMs, totalBytes, blockCount, bmId)
+      val expectedMessage = s"Slow shuffle block fetch detected: Fetching blocks took longer " +
+        s"than expected ( blockCount = $blockCount , totalBlockSize = $totalBytes , " +
+        s"blockManagerId = BlockManagerId(execID, hostname, 1234, None) , " +
+        s"durationMs = $elapsedMs , " +
+        s"transferBps = ${(totalBytes.toDouble / elapsedMs * 1000).toLong} )\n"
+      assert(expectedMessage === logCapture.getOutput)
+
+      // Fetch with a read that is slightly faster than the threshold
+      logCapture.clearOutput()
+      fetcher.logFetchIfSlow(1000 * totalBytes / thresholdBps - 1, totalBytes, blockCount, bmId)
+      assert("" === logCapture.getOutput)
+
+      // When the number of bytes is too small, even though the transfer rate is slow, the
+      // thresholdMillis will not be satisfied
+      val smallBytes = 10
+      fetcher.logFetchIfSlow(1000 * smallBytes / thresholdBps + 1, smallBytes, blockCount, bmId)
+      assert("" === logCapture.getOutput)
+    } finally if (logCapture != null) {
```

Comment:
`logCapture` will never be null here right? Is this check necessary?

Reply:
👍 Removed